php/class-sample.php

```diff
@@ -19,15 +21,24 @@ class Sample {
 	 */
 	public $plugin;
 
+	/**
+	 * Sample class from a sub-namespace
+	 *
+	 * @var SubSample
+	 */
+	public $sub_sample;
+
 	/**
 	 * Constructor.
 	 *
 	 * @access public
 	 *
-	 * @param Plugin $plugin Plugin instance.
+	 * @param Plugin    $plugin     Plugin instance.
+	 * @param SubSample $sub_sample SubSample instance.
 	 */
-	public function __construct( Plugin $plugin ) {
-		$this->plugin = $plugin;
+	public function __construct( Plugin $plugin, SubSample $sub_sample ) {
```

Comment:
Why inject it as a dependency?

Reply:
No reason in particular, just for demonstration purposes. This Sample class isn't there to stay anyway when somebody picks up this package.

Reply:
@derekherman Removed that "demo". Now it's just a standalone class with no real use. It has a small test though which demonstrates that it works.